common/persistence/nosql/nosqlplugin/cassandra/tasks.go

```diff
@@ -387,23 +387,25 @@ PopulateTasks:
 			continue
 		}
 
-		// Extract the TTL value
-		ttlValue, ttlExists := task["ttl"]
-
-		// Check if TTL is null or an integer
-		var ttl *int
-		if ttlExists && ttlValue != nil {
-			if ttlInt, ok := ttlValue.(int); ok {
-				ttl = &ttlInt // TTL is an integer
-			}
-		}
+		// TODO: no usage of ttl
+		//// Extract the TTL value
+		//ttlValue, ttlExists := task["ttl"]
+		//
+		//// Check if TTL is null or an integer
+		//var ttl *int
+		//if ttlExists && ttlValue != nil {
+		//	if ttlInt, ok := ttlValue.(int); ok {
+		//		ttl = &ttlInt // TTL is an integer
+		//	}
+		//}
 
 		t := createTaskInfo(task["task"].(map[string]interface{}))
 		t.TaskID = taskID.(int64)
 
-		if ttl != nil {
-			t.Expiry = db.timeSrc.Now().Add(time.Duration(*ttl) * time.Second)
-		}
+		// TODO: removing this, because there is no usage of it
```

Comment:
same here

Reply:
Another issue at this low-level layer is that TTL expiry **computation** is happening here, whereas I understand that we should remove any **computation** from db layer entirely. Moving this logic to upstream led nowhere, as TTL isn't used there either.

Since the task is to remove `timeSrc` from `db` and we need to discuss this matter later, I could replace the current logic with `time.Now()` and keep the TODO.